### DISCUSSION TITLE
deepPartial is deprecated. What to use instead?

### DISCUSSION THREAD

**Discussion by codepilotsf** [colinhacks/zod#2527] [Q&A]
According to VS Code, we should no longer use deepPartial because it has been deprecated.
https://www.codepilot.com/dropshare/Screen-Shot-2023-06-22-13-45-37.61.png

So if I need to partially validate a schema with a nested object, what should I use instead?

**Comment by thiagoanselmo** [comment]
Any news about this?

**Reply by hyperupcall** [reply]
PLEASE don't bump threads on GitHub. It's disrespectful to the maintainers time and it's disrespectful to the people that have subscribed to this thread. If you have nothing to add, then please don't say anything.

**Reply by seanyasno** [reply]
so no fix?

**Comment by CanRau** [comment]
There's an open issue https://github.com/colinhacks/zod/issues/2854 with potential solutions/alternatives

**Comment by amirfarzamnia** [comment]
🚀 **Solution Available: `zod-deep-partial`**

I've put together a solution to address the current issue: a fully type-safe utility that deeply makes any Zod schema `partial`, including nested objects, arrays, and all Zod types.

### ✅ Features

* **Full Type-Safety** – Built with TypeScript from the ground up.
* **Supports All Zod Types** – Handles complex, nested schemas correctly.
* **Zero Dependencies** – Lightweight (\~10KB).
* **Focused Utility** – Does one thing: deeply partializes Zod schemas.

Get it on npm:

[https://www.npmjs.com/package/zod-deep-partial](https://www.npmjs.com/package/zod-deep-partial)

Source Code:

[https://github.com/amirfarzamnia/zod-deep-partial](https://github.com/amirfarzamnia/zod-deep-partial)

Documentation:

[https://amirfarzamnia.github.io/zod-deep-partial/](https://amirfarzamnia.github.io/zod-deep-partial/)

It's simple to use: just import the function and apply it to your Zod schema.

Contributions and feedback are welcome!
